src/EtherBalance.jsx

```diff
@@ -0,0 +1,18 @@
+const React = require('react');
+const { BigNumber } = require('bignumber.js');
+const { ReactiveComponent } = require('oo7-react');
+
+// keeps balance of ether up to date
+class EtherBalance extends ReactiveComponent {
+	constructor () { super(['balance']); }
+
+	render () {
+		if (typeof this.state.balance === 'undefined') { return (<span>-</span>); }
+		let ethdiv = new BigNumber('1e+18');
```

Comment:
Putting comment here, but relevant in all files.

Do a CTRL+F on 'let' in your workspace, and replace them by 'const', *except* where you know that the variable can change value.

Or else we'll wonder when it's gonna change value.